src/toolchain_cmd.go

```diff
@@ -414,9 +414,9 @@ func installGoToolchain() error {
 		return skippedToolchain{toolchain, "no GOPATH set (assuming Go is not installed and you don't want the Go toolchain)"}
 	}
 
-	srclibpathDir := filepath.Join(strings.Split(srclib.Path, ":")[0], toolchain) // toolchain dir under SRCLIBPATH
+	srclibpathDir := filepath.Join(filepath.SplitList(srclib.Path)[0], toolchain) // toolchain dir under SRCLIBPATH
 
-	if err := os.MkdirAll(filepath.Dir(srclibpathDir), 0700); err != nil {
```

Comment:
I'm not sure why this `filepath.Dir` was here. On Windows it actually removes the toolchain dir and makes it so the code doesn't make any sense. We immediately try to symlink a folder that we didn't create.

I'm thinking on *nix there's a trailing slash and this just ends up trimming the trailing slash?

This happens on line 551 too.